Source/JavaScriptCore/runtime/TemporalPlainYearMonth.cpp

```diff
@@ -118,6 +118,95 @@ String TemporalPlainYearMonth::toString(JSGlobalObject* globalObject, JSValue op
     return ISO8601::temporalYearMonthToString(m_plainYearMonth, calendarName);
 }
 
+// https://tc39.es/proposal-temporal/#sec-temporal.plainyearmonth.from
+// https://tc39.es/proposal-temporal/#sec-temporal-totemporalyearmonth
+// optionsValue may be undefined, which is treated as the absence of an options argument
+TemporalPlainYearMonth* TemporalPlainYearMonth::from(JSGlobalObject* globalObject, JSValue item, JSValue optionsValue)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    // Handle string case first so that string parsing errors (RangeError)
+    // can be thrown before options-related errors (TypeError);
+    // see step 4 of ToTemporalYearMonth
+    TemporalPlainYearMonth* result;
+    bool isString = item.isString();
+
+    if (isString) {
+        auto string = item.toWTFString(globalObject);
+        RETURN_IF_EXCEPTION(scope, { });
+        result = TemporalPlainYearMonth::from(globalObject, string);
```

Comment:
The local variable `isString` doesn’t do us any good here. Also, the `JSValue` member function `getString` is designed for this kind of use and is more efficient than calling `isString` and then `toWTFString`. And `result` should be scoped inside the if statement.
```suggestion
    auto string = item.getString(globalObject);
    RETURN_IF_EXCEPTION(scope, { });
    if (!string.isNull()) {
        auto* result = TemporalPlainYearMonth::from(globalObject, string);
```